pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -61,19 +63,71 @@ func NewReceiverService(
 	}
 }
 
-func (rs *ReceiverService) canDecrypt(ctx context.Context, user identity.Requester, name string) (bool, error) {
-	receiverAccess := false // TODO: stub, check for read secrets access
-	eval := accesscontrol.EvalPermission(accesscontrol.ActionAlertingProvisioningReadSecrets)
-	provisioningAccess, err := rs.ac.Evaluate(ctx, user, eval)
+func (rs *ReceiverService) shouldDecrypt(ctx context.Context, user identity.Requester, name string, reqDecrypt bool) (bool, error) {
+	// TODO: migrate to new permission
+	eval := accesscontrol.EvalAny(
+		accesscontrol.EvalPermission(accesscontrol.ActionAlertingReceiversReadSecrets),
+		accesscontrol.EvalPermission(accesscontrol.ActionAlertingProvisioningReadSecrets),
+	)
```

Comment:
I would love authz logic be in a special methods in ngalerts/accesscontrol package. So we will not have to spread knowledge about actions, scopes etc. Not a blocker for this PR.